From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Tue, 29 Apr 2025 16:16:10 +0000
Subject: [PATCH] Disable tab group sync settings UI by default

Adapted from GrapheneOS Vanadium patch 0060-Disable-tab-group-sync-settings-UI-by-default.patch for the Vandium browser port.

---
 .../chrome/browser/tasks/tab_management/TabsSettings.java | 5 ++++-
 chrome/browser/flags/android/chrome_feature_list.cc       | 1 +
 .../chromium/chrome/browser/flags/ChromeFeatureList.java  | 2 ++
 components/saved_tab_groups/public/features.cc            | 8 ++++++++
 components/saved_tab_groups/public/features.h             | 2 ++
 5 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/components/saved_tab_groups/public/features.cc b/components/saved_tab_groups/public/features.cc
index 32b35ac6fdb25..c4b77391b6b6c 100644
--- a/components/saved_tab_groups/public/features.cc
+++ b/components/saved_tab_groups/public/features.cc
@@ -37,6 +37,14 @@ BASE_FEATURE(kTabGroupSyncDelegateAndroid,
              "TabGroupSyncDelegateAndroid",
              base::FEATURE_ENABLED_BY_DEFAULT);
 
+// Feature flag to disable auto-open of saved tab groups. Note that the
+// settings page for auto open will still be visible, and when user is allowed
+// to change. However the written pref from the user selection will not be
+// honored. This feature flag should be used only in case of an emergency.
+BASE_FEATURE(kTabGroupSyncAutoOpenKillSwitch,
+             "TabGroupSyncAutoOpenKillSwitch",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
 // Feature flag to restrict download on synced tabs if the navigation is
 // triggered without attention..
 BASE_FEATURE(kRestrictDownloadOnSyncedTabs,
diff --git a/components/saved_tab_groups/public/features.h b/components/saved_tab_groups/public/features.h
index 473045460df16..daf91efc8509b 100644
--- a/components/saved_tab_groups/public/features.h
+++ b/components/saved_tab_groups/public/features.h
@@ -18,6 +18,8 @@ BASE_DECLARE_FEATURE(kTabGroupSyncServiceDesktopMigration);
 
 BASE_DECLARE_FEATURE(kTabGroupSyncDelegateAndroid);
 
+BASE_DECLARE_FEATURE(kTabGroupSyncAutoOpenKillSwitch);
+
 BASE_DECLARE_FEATURE(kRestrictDownloadOnSyncedTabs);
 
 BASE_DECLARE_FEATURE(kUseAlternateHistorySyncIllustration);
